Initial commit

cs-tushar-kanade opened the pull request:
Initial commit

cs-tushar-kanade committed:
Initial commit

Initial commit

cs-pooja-singh commented on csv-data-management/info.json:
```diff
@@ -0,0 +1,68 @@
+        {
+            "title": "Extract Data from CSV",
+            "operation": "extract_data_from_csv",
+            "description": "This action will extract data from CSV based on column names provided. User can select option to dedupicate rows based on columns",
```
deduplicate

cs-pooja-singh commented on csv-data-management/info.json:
```diff
@@ -0,0 +1,68 @@
+                 "title": "Type",
+                  "type": "select",
+                  "options": [
+                             "Attachment ID",
```
Attachment IRI? to keep it consistent

cs-tushar-kanade replied:
Done

cs-pooja-singh commented on csv-data-management/info.json:
```diff
@@ -0,0 +1,68 @@
+                    "required": false,
+                    "visible": true,
+                    "editable": true,
+                    "value": ""
```
is it a comma separated list? an example with a 'placeholder' key can be useful

cs-tushar-kanade replied:
Yes, its comma separated . I will add placeholder

cs-pooja-singh commented on csv-data-management/info.json:
```diff
@@ -0,0 +1,68 @@
+            "category": "utilities"
+        }
+    ],
+    "forked_from": false,
```
these two keys were not removed while exporting the connector?

cs-pooja-singh commented on csv-data-management/info.json:
```diff
@@ -0,0 +1,68 @@
+    "version": "1.0.0",
+    "label": "CSV Data Management",
+    "description": "",
+    "publisher": "",
```
Fortinet?

cs-tushar-kanade replied:
Done

cs-pooja-singh commented on csv-data-management/info.json:
```diff
@@ -0,0 +1,68 @@
+    "icon_large_name": "fortisoar_logo.png",
+    "category": "",
+    "configuration": {
+        "fields": []
```
remove this completely. it will, not otherwise show the message that the connector doesn't require a configuration

cs-tushar-kanade replied:
Done

cs-pooja-singh commented on csv-data-management/connector.py:
```diff
@@ -0,0 +1,23 @@
+
+class Csvdatamanagement(Connector):
+
+    def dev_execute(self, config, operation, params, *args, **kwargs):
```
let's remove this function?

cs-tushar-kanade replied:
Done.

cs-pooja-singh commented on csv-data-management/extract_data_from_csv.py:
```diff
@@ -0,0 +1,69 @@
+            deDupValuesOn = deDupValuesOn.split(",")
+            df.drop_duplicates(subset=[deDupValuesOn], keep='first')
+
+        smaller_datasets = np.array_split(df, 20)
```
why split when we need to join again?

cs-tushar-kanade replied:
The number of records returning is too large at times when CSV file is large one , hence to prevent playbook for crashing , we can let user loop returned records over a number of smaller data set. But we now have bulk insert feed , may be I will be take it as an option in operation . Please advice @cs-pooja-singh

cs-suyog-jinturkar commented on csv-data-management/extract_data_from_csv.py:
```diff
@@ -0,0 +1,69 @@
+        logger.info("file from code:" + file_path)
+        if params.get('columnNames'):
+            columnNames = params.get('columnNames')
+            columNames = columNames.split(",")
```
spelling mistake? columNames Vs columnNames

cs-suyog-jinturkar replied:
And we are not using columNames anywhere in if block

cs-tushar-kanade replied:
typo corrected

cs-suyog-jinturkar commented on csv-data-management/extract_data_from_csv.py:
```diff
@@ -0,0 +1,69 @@
+            columNames = columNames.split(",")
+            df = pd.read_csv('{}'.format(file_path), delimiter=',', encoding="utf-8-sig")[[columNames]]
+        else:
+            df = pd.read_csv('{}'.format(file_path), delimiter=',', encoding="utf-8-sig")
```
what will else do? one liner comment will help here

cs-tushar-kanade replied:
Done , added comments for both read call of csv

cs-tushar-kanade replied:
Done , added comments for both read call of csv

cs-tushar-kanade replied:
Done , added comments for both read call of csv

cs-suyog-jinturkar commented on csv-data-management/extract_data_from_csv.py:
```diff
@@ -0,0 +1,69 @@
+
+    except Exception as Err:
+        logger.error('Error in extract_data_from_csv(): %s' % Err)
+        logger.exception('Error in extract_data_from_csv(): %s' % Err)
```
why we are using both logger.error and logger.exception?

cs-tushar-kanade replied:
Need to check if MP framework is adding. I will go through workspace in product to check these things

cs-suyog-jinturkar commented on csv-data-management/extract_data_from_csv.py:
```diff
@@ -0,0 +1,69 @@
+            else:
+                raise ConnectorError('Invalid File IRI {0}'.format(value))
+    except Exception as err:
+        logger.info('handle_params(): Exception occurred {0}'.format(err))
```
logs are not consistent in this function.

logger.info('handle_params():
logger.info('file.......

cs-suyog-jinturkar replied:
Why logger.info and not logger.error or logger.exception?

cs-tushar-kanade replied:
debug logs , will remove once final code review is done. Adding comments to identify these logs statements

cs-tushar-kanade replied:
This code is from VT

cs-tushar-kanade committed:
Review feedback

Partial review feedback incorporated

cs-tushar-kanade committed:
correcting JSON

correcting JSON

cs-tushar-kanade committed:
Updating current operation

Updating current operation
1. Add option to skip rows
2. Add option to read file with no column headers
3. PR feedback

cs-tushar-kanade committed:
Complete Refactoring

Complete Refactoring of code

Added one more operation
Update info.json

cs-tushar-kanade committed:
making field mandatory

making field mandatory  for merging

cs-tushar-kanade committed:
Correcting return result

Correcting return result , no need to json dumps as result is in json format and we need it that way to consume in playbook

cs-tushar-kanade committed:
Handling condition for invalid JSON

When json as NaN value , it breaks connector framework. Hence replacing NaN with string N/A

cs-tushar-kanade committed:
Fixing coding error

Fixing coding error

cs-tushar-kanade committed:
fixing coding error

fixing coding error

cs-tushar-kanade committed:
Optimising RAM usage

Reading in chunks to reduce impact on usage of RAM

cs-tushar-kanade committed:
Updating Logic

Adding logic to check if uploaded file is CSV and has headers or not. Also to check if column has headers

cs-tushar-kanade committed:
Updating chunksize

Updating chunk size and will skip lines from csv file which are erroneous

cs-tushar-kanade committed:
PR review

PR review changes

cs-tushar-kanade committed:
PR Feedback

PR Feedback

cs-tushar-kanade committed:
removing medium icon

removing medium icon
